### PR TITLE
Reducing of tables to tables not using cellpadding or cellspacing attributes

### PR DESCRIPTION
**Pull request by shinze**
Here is a sample table from CKEditor a popular editor :
http://codepen.io/anon/pen/BjogzZ?editors=110

The problem is that the user cannot change the aspect of this table with anything but these two attributes. The idea is to reduce the scope of tables and td, th to tables not using these attributes.

### CODE REVIEW

File: normalize.css [hunk 1]
```diff
@@ -415,10 +415,13 @@ optgroup {
 
 table {
   border-collapse: collapse;
+}
+
+table:not[cellspacing] {
```
Comment (cvrebert) [inline normalize.css]: I think your syntax is wrong here.
Presumably you want `table:not([cellspacing])`.
Same problem below too.

**Comment by cvrebert** [comment]
Note: Both of those attributes are deprecated.

**Comment by jonathantneal** [comment]
Is reducing table cell spacing a browser normalization? Do browsers implement this spacing differently? If so, is no spacing even the best experience?

If our CSS padding is just an opinionated reset, I recommend we remove it from Normalize.css. Let [other](https://github.com/twbs/bootstrap/blob/v4-dev/scss/_tables.scss#L10-L16) [libraries](https://github.com/10up/sanitize.css/blob/master/sanitize.css#L118-L124) handle the opinions.

**Comment by battaglr** [comment]
As @cvrebert said, the attributes [cellpadding](https://www.w3.org/TR/html5/obsolete.html#attr-table-cellpadding) and [cellspacing](https://www.w3.org/TR/html5/obsolete.html#attr-table-cellspacing) are officially [obsolete](https://www.w3.org/TR/html5/obsolete.html#non-conforming-features). IMO this reason alone should be enough to close this PR.

**Comment by shinze** [comment]
Both attributes are obsolete in html 5 flavor but is this a prerequisite?
@jonathantneal is asking the right questions here, is this a normalization or a reset feature ?

**Comment by battaglr** [comment]
@shinze: you are right, I totally overlooked @jonathantneal comment, and he has raised a very good question.

As far as I can tell, seems that `table` rules are fairly opinionated, since [in the comments there is not explicit reference to browsers rendering these properties differently](https://github.com/necolas/normalize.css/blob/master/normalize.css#L422) by default —as almost any other declaration does.

In #354 we were discussing about removing `border-spacing` —the "CSS equivalent" of the attribute `cellspacing`— as it applies only when `border-collapse` is `separate`. If we maintain `border-collapse: separate` the rule [`table:not[cellspacing]`](https://github.com/shinze/normalize.css/blob/master/normalize.css#L420) (sic) is not going to have any visible effect, since `cellspacing` it's also applied only when `border-collapse` is `separate` —here is a [test case](https://jsfiddle.net/battaglr/xjxhqb77/).

With that out of the way, as @jonathantneal implied the discussion should be focused in whether we should use `border-collapse: collapse` —[being `separate` the initial value](https://www.w3.org/TR/CSS2/tables.html#propdef-border-collapse)— and `padding: 0` —which initial value is apparently `1px` but didn't find documentation about it— for tables.

As I said before, IMO those are in fact opinionated styles, and probably go against the original idea of Normalize.css if we take the project description to the letter: "it precisely targets only the styles that need normalizing".

If we remove those declarations, it will have a huge impact on websites relying on those values, thus we probably will have to bump to v4.

**Comment by shinze** [comment]
@battaglr In my opinion all this block about tables should be "simply" removed. But as you say it has a huge impact and will bump normalize to a V4.

It has to be tested but these feature are not really well documented (could not find any good source). Here is a start for a test case of this : http://codepen.io/shinze/pen/JGpyNV.

**Comment by jonathantneal** [comment]
I'll test that page and do further research as well.

Also, I'd like to move the bumping of Normalize to a separate discussion.

**Comment by shinze** [comment]
Here is a better test case : http://codepen.io/shinze/pen/mVXqLg.

Tried it on a "main" browsers can't spot any difference : https://www.browserstack.com/screenshots/6a5bd42dcceb926f5a66f5033c2b36f2a50d2ff7

**Comment by battaglr** [comment]
Just to be clear, IMHO if we decide that these styles are opinionated, I'll remove `border-collapse: collapse` as well —meaning all `table`, `th` and `td` will be gone. Not because those are not good defaults, but because those values don't need normalization.

I guess this is not an easy call, @jonathantneal. If you need any help testing in some devices or browsers let us know.

**Comment by shinze** [comment]
About border spacing.

W3C wiki says 'border-spacing' default value is 0
https://www.w3.org/wiki/CSS/Properties/border-spacing

This document showing the recommended default CSS by W3C shows a value of '2px'.
https://www.w3.org/TR/CSS2/sample.html

Applied by :
* WebKit : http://trac.webkit.org/browser/trunk/Source/WebCore/css/html.css
* Explorer : http://www.iecss.com
* Mozilla: http://hg.mozilla.org/mozilla-central/file/tip/layout/style/html.css

**Comment by jonathantneal** [comment]
Thanks @shinze. It seems like what remains is to test all browsers to see if 1. they follow the spec, and / or 2. if they match each other consistently across our whole support stack.

@battaglr, would you have the time or resources to test these default table values across browsers? If not, I will try to make it a priority. Thanks again!

For reference, these changes were made to [add table element normalization](https://github.com/necolas/normalize.css/commit/a27246fcabe7a5b57e08341aac36b81c287cac71) though they were possibly [convenience changes](https://github.com/necolas/normalize.css/commit/37d1c1ef8ba2ddd5708bfc93083cda964711e6aa).

One thing we can do moving forward is always tag a browser and a version in the comment.

**Comment by battaglr** [comment]
Thanks for the references.

Sure, I can do that. Just to be certain I understood correctly, the idea is to test `table`, `th` and `td` on different browsers without any kind of normalization?

**Comment by jonathantneal** [comment]
@battaglr, yes. If there are no differences then we know we can remove these lines. If there are differences, then we can properly identify them in a comment.

**Comment by jonathantneal** [comment]
@battaglr, the more I read about this issue, the more I’m expecting table styles to appear consistent without Normalize.css. Have you had a chance to test or would you like me to?

**Comment by battaglr** [comment]
@jonathantneal: sadly has been a busy week! I'm on it now, and I'll share the results ASAP.

**Comment by battaglr** [comment]
Here are [the results](https://gist.github.com/battaglr/2c3c53aff8043fbfbb19) of the [test case](https://jsfiddle.net/battaglr/ha29xato/) on different browsers and platforms.

As far as I can see, it's safe to remove the current normalizations for table elements, since all browsers seems to render them consistently.

That being said, would be really nice to have some feedback from the community and even better if someone else could test this as well.

---

*Keep in mind that the listed results listed are just the ones obtained using BrowserStack. I also tested on my Mac and Android and the results were the expected ones.*

**Comment by jonathantneal** [comment]
Wow. First, great job getting those results, @battaglr. We need to talk about adding something like to the Normalize.css release process with the test files. I wanted to use BrowserStack for this, but I was taken aback when they did not support the latest browsers.

So, back to your tests; in testing 72 devices, every browser — every modern browser and every older browser going back to Internet Explorer 6 — is consistent. While the `border-spacing` property did not exist before Internet Explorer 8, I had seen the same results pinging `currentColor` (an IE proprietary property on nodes that is somewhat similar to getComputedStyle).

What sort of feedback would you be looking for from the community? Is there a particular scenario in mind you may be thinking of?

**Comment by battaglr** [comment]
@jonathantneal, hehe, I know, those were a lot of tests... I guess I let myself be carried away by curiosity.

Regarding further feedback from the community, there is no particular scenario that worries me, and to be really honest, I think it's just that those styles have been the "default" for me —and many others— since [Eric Meyer's CSS reset v1](http://meyerweb.com/eric/tools/css/reset/reset200802.css) —that was ~8 years ago!

Leaving the nostalgia aside, the reasoning to remove those styles from *Normalize.css* are pretty clear, so as I said in previous comments, I believe that removing them is the right call.

*What do you think about creating a new issue to discuss possible tests for the release process, or even how to properly create test cases and/or test results? I think this will make maintenance easier.*

**Comment by jonathantneal** [comment]
I totally endorse creating a new issue to properly test changes. It would be huge, and I want it before Normalize v4. @lukewatts is looking into a solution for this as well in 10up/sanitize.css#59

**Comment by jonathantneal** [comment]
This issue and pull is resolved by our conclusion to remove the opinionated table resets in 02c5c7a. Also, see #506

**Pull request closed without merging**